packages/runtime/src/components/page/PageHead.tsx

```diff
@@ -0,0 +1,188 @@
+  const isIndexingBlocked = baseLocalizedPage?.seo.isIndexingBlocked ?? page.seo.isIndexingBlocked
+
+  const fontFamilyParamValue = useMemo(() => {
+    if (site == null) {
+      return page.fonts
+        .map(({ family, variants }) => {
+          return `${family.replace(/ /g, '+')}:${variants.join()}`
+        })
+        .join('|')
+    }
+
+    return site.googleFonts.edges
+      .filter((edge): edge is NonNullable<typeof edge> => edge != null)
+      .map(({ activeVariants, node: { family, variants } }) => {
+        const activeVariantSpecifiers = variants
+          .filter(variant =>
+            activeVariants.some(activeVariant => activeVariant.specifier === variant.specifier),
+          )
+          .map(variant => variant.specifier)
+          .join()
+
+        return `${family.replace(/ /g, '+')}:${activeVariantSpecifiers}`
+      })
+      .join('|')
+  }, [site, page])
+
+  const headElementsFragment = (
+    <>
+      <style>
+        {`
+      html {
+        font-family: sans-serif;
+      }
+      div#__next {
+        overflow: hidden;
+      }
+    `}
+      </style>
+
+      {favicon && <link rel="icon" type={favicon.mimetype} href={favicon.publicUrl} />}
+
+      {canonicalUrl && <link rel="canonical" href={canonicalUrl} />}
+
+      {isIndexingBlocked && <meta name="robots" content="noindex" />}
+
+      {title && (
+        <>
+          <title>{title}</title>
+          <meta property="og:title" content={title} />
+          <meta name="twitter:title" content={title} />
+          <meta itemProp="name" content={title} />
+        </>
+      )}
+
+      {description && (
+        <>
+          <meta name="description" content={description} />
+          <meta property="og:description" content={description} />
+          <meta name="twitter:description" content={description} />
+          <meta itemProp="description" content={description} />
+        </>
+      )}
+
+      {keywords && <meta name="keywords" content={keywords} />}
+
+      {socialImage && (
+        <>
+          <meta property="og:image" content={socialImage.publicUrl} />
+          <meta property="og:image:type" content={socialImage.mimetype} />
+          <meta name="twitter:image" content={socialImage.publicUrl} />
+          <meta name="twitter:card" content="summary_large_image" />
+          <meta itemProp="image" content={socialImage.publicUrl} />
+        </>
+      )}
+
+      {fontFamilyParamValue !== '' && (
+        <link
+          rel="stylesheet"
+          href={`https://fonts.googleapis.com/css?family=${fontFamilyParamValue}&display=swap`}
+        />
+      )}
+
+      {headSnippets.map(snippetToElement).map(children =>
+        Children.map(children, child => {
+          if (typeof child === 'string') return child
+
+          if (VALID_HEAD_ELEMENT_TYPES.includes(child.type as string)) return child
+
+          return null
+        }),
+      )}
```

Comment:
I'm worried that the [nesting](https://nextjs.org/docs/pages/api-reference/components/head#use-minimal-nesting) here will be a problem. Have we tested for this?

Reply:
Confirming that nesting is not an issue for snippets in rendered in the next/head